src/experimental/safe_html_transform/transforms/exp_safe_html.py

```diff
@@ -0,0 +1,320 @@
+                                      'any tag.',
+                                      ('tag', 'value')),
+            'style_whitelist': ('list',
+                                'style_whitelist',
+                                'These CSS styles are allowed in style ' +
+                                'attributes.'),
+            'class_blacklist': ('list',
+                                'class_blacklist',
+                                'These class names are not allowed in ' +
+                                'class attributes.'),
+            'remove_javascript': ("int",
+                                  'remove_javascript',
+                                  '1 to remove javascript attributes that ' +
+                                  'begin with on (e.g. onClick) and ' +
+                                  'attributes where the value starts with ' +
+                                  '"javascript:" (e.g. ' +
+                                  '<a href="javascript:function()". This ' +
+                                  'does not effect <script> tags. 0 to ' +
+                                  'leave the attributes.'),
+            'disable_transform': ("int",
+                                  'disable_transform',
+                                  'If 1, nothing is done.'),
+            }
+
+        self.config.update(kwargs)
+
+        if name:
+            self.__name__ = name
+
+    def name(self):
+        return self.__name__
+
+    def __getattr__(self, attr):
+        if attr == 'inputs':
+            return self.config['inputs']
+        if attr == 'output':
+            return self.config['output']
+        raise AttributeError(attr)
+
+    def convert(self, orig, data, **kwargs):
+        # note if we need an upgrade.
+        if 'disable_transform' not in self.config:
+            log(logging.ERROR, 'PortalTransforms safe_html transform needs '
+                'to be updated. Please re-install the PortalTransforms '
+                'product to fix.')
+
+        # if we have a config that we don't want to delete
+        # we need a disable option
+        if self.config.get('disable_transform'):
+            data.setData(orig)
+            return data
+
+        if orig == "" or orig == "<html></html>" or orig == "<html />" or orig == "<html/>":
+            safe_html = str("")
+            data.setData(safe_html)
+
+        else:
+            # append html tag to create a dummy parent for the tree
+            html = "<html>%s</html>" % orig
+            NASTY_TAGS = frozenset(['style', 'script', 'object', 'applet', 'meta', 'embed'])  # noqa
+            cleaner = HTMLParser(kill_tags=NASTY_TAGS, page_structure=False, safe_attrs_only=False)
+            safe_html = fragment_fromstring(cleaner.clean_html(html))
+            safe_html2 = ""
+            for i in range(len(safe_html)):
+                safe_html1 = tostring(safe_html[i])
+                safe_html2 = safe_html2 + safe_html1
```

Comment:
@lentinj  Because to avoid the extra "body" tag as if we don't fragment the string and directly pass the lxml created tree then it will append an extra <body> tag to the tree and what I tried there is I broke the strings into fragments and then again converted them into string and then add them and then pass them so it didn't append the body tag. Yeah here I broke the tree :/ 

I guess we can do one more thing that we can keep the tree structure created by lxml and then remove the extra appended body tag also as we did for the <html> tag , is that sounds good ?